test/src/test/java/jenkins/model/JenkinsTest.java

```diff
@@ -617,4 +618,40 @@ public void versionIsSavedInSave() throws Exception {
         VersionNumber nullVersion = Jenkins.getStoredVersion();
         assertNull(nullVersion);
     }
+
+    @Issue("JENKINS-46911")
+    @WithoutJenkins
+    @Test
+    public void testCheckGoodName() throws Exception{
+        String[] illegalNames = {"ab\\c", "abc/", "ab/c", " ", "   ", "", "..", ".", "?", "*", "6%",
+                "x!", "-@", "#", "$", "^", "&", "|", "<", ">", "[", "]", ":", ";", "../.", null};
+        String[] legalNames = {"abc", "a bc", "abc ", " abc", "a.bc", ". ."};
+        String msg = "checkGoodNameTest found a legal name. Please check.";
+        int exceptions = 0;
+        int goodNames = 0;
+
+        for (String name : illegalNames){
+            try {
+                Jenkins.checkGoodName(name);
+                goodNames++;
+            } catch (Failure e){
+                exceptions++;
+            }
+        }
+        assertEquals(msg, illegalNames.length, exceptions);
+        assertEquals(0, goodNames);
+
+        exceptions = 0;
+        goodNames = 0;
+        for (String name : legalNames){
+            try {
+                Jenkins.checkGoodName(name);
+                goodNames++;
+            } catch (Failure e){
+                exceptions++;
+            }
+        }
+        assertEquals(msg, 0, exceptions);
```

Comment:
see comment on other test, you can store the name in an list and then assert the size of that list is zero and get a view of what failed. rather than 2 things failed.